### DISCUSSION TITLE
Contatempo di funzionamento

### DISCUSSION THREAD

**Discussion by Silvano2161** [indomus/forum#990] [HUB: Home Assistant]
Ciao a tutti, questa è la mia prima richiesta, sono nuovo sia di questo sito che in generale del mondo della domotica.

Mi sono avvicinato a questa realtà perché incuriosito dei nuovi metodi di gestione degli oggetti casalinghi che aiutano a controllare l'andamento dei consumi domestici.

Sono un perito elettronico, quindi con una discreta conoscenza del mondo elettrico e dei suoi fenomeni, ma abbastanza incompetente sui misteri della programmazione e di informatica so quanto basta per accendere un PC ed andare in rete alla ricerca di informazioni varie.

Fatte le dovute presentazioni passo alla descrizione del mio impianto:

Ho installato su un miniPC tipo netbook che avevo da qualche tempo messo in disparte il S.O. Ubuntu 20.04 LTS sul quale, grazie appunto ad info reperite in giro, ho installato Home Assistance (tanto per provarlo ed eventualmente in futuro passare ad un Raspberry dedicato).
Il mio impianto domotico consta di più oggetti Shelly 2.5 e Shelly 1PM che mi permettono di accendere/spegnere luci e/o utilizzatori elettrici e di monitorarne il consumo elettrico, inoltre ho uno Shelly EM che con le sue sonde amperometriche mi controlla sia l'ingresso dell'energia dal contatore Enel che la produzione del mio impianto fotovoltaico.
Per il riscaldamento utilizzo sia una pompa di calore, che con l'ausilio di H.A. ed un Tado° Climatizzazione V3+ conto di far partire quando ho energia sufficiente dai pannelli solari, che una classica caldaia a gas gestita da due termostati, uno per il piano terra ed uno per il primo piano con due elettrovalvole che, quando aperte, comandano in or l'avviamento della caldaia.
A questo scopo ho utilizzato un ulteriore modulo Shelly 1 che riceve in parallelo le posizioni di "Aperto" delle elettrovalvole (per cui ne basta una aperta) e a sua volta ripete il comando tramite il suo contatto pulito alla caldaia.

Fin qui tutto bene, grazie ad H.A. visualizzo il tutto regolarmente ed intanto sto studiando qualche metodo per gestire le cose  in modo interattivo (al momento mi accontento di monitorare da un unico pannello quelli che fino a poco fa erano oggetti separati e fini a se stessi).
Ringrazio per questo voi e questo sito per avermi portato a conoscenza di questo strumento di gestione.

Ora tutta questa introduzione mi porta finalmente al motivo della mia richiesta:

Lo Shelly 1 che comanda la caldaia mi restituisce un grafico temporale dei momenti in cui va in "ON" ossia di quando la caldaia è accesa,
Molto bello e coreografico, ma mi è sorta la voglia di sapere a fine giornata per quanto tempo complessivo la caldaia è stata accesa.
Siccome insieme al grafico è presente una lista in cui compaiono tutti i momenti corredati di marca oraria dei cambiamenti stato ON/OFF questi dati sono già presenti da qualche parte in qualche database, per cui:

E' possibile in qualche modo far si che un programma? - Script? - o come diavolo si potrebbe chiamare una lista di istruzioni "conti" i tempi in cui lo stato di uscita è rimasto in "ON" e li sommi per avere un totale giornaliero?

Scusate la prolissità della descrizione, se avete avuto il buon cuore di giungere fino a qui sappiate che avete i miei più vivi ringraziamenti.

Silvano

**Comment by xyzmauro** [comment]
Ciao, questo sensor fa proprio quel che cerchi tu
https://www.home-assistant.io/integrations/history_stats/

**Comment by Silvano2161** [comment]
Grazie!!

Ho visto che ci sono dei dati da scrivere, al momento non ho idea di dove andarli a scrivere, ma prima o poi riuscirò a capirlo, intanto è una buona base di partenza.

Se e quando ci riuscirò, vi terrò informati del successo ottenuto.

Silvano
